rand_distr/src/zipf.rs

```diff
@@ -0,0 +1,322 @@
+// Copyright 2021 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! The Zeta and related distributions.
+
+use num_traits::Float;
+use crate::{Distribution, Standard};
+use rand::{Rng, distributions::OpenClosed01};
+use core::fmt;
+
+/// Samples floating-point numbers according to the zeta distribution.
+///
+/// The zeta distribution is a limit of the [`Zipf`] distribution.
+///
```

Comment:
We can also add the actual algorithm implemented. 
Reference: [[Devroye86](https://doi.org/10.1007/978-1-4613-8643-8)] Section 6.1, page 551.

Reply:
I linked to it in the code.

Reply:
In [StandardNormal](https://docs.rs/rand_distr/0.4.1/rand_distr/struct.StandardNormal.html) the algorithm is visible in the documentation. 
Personally, I like that, is there a unified policy for this?

Reply:
That's a good point, I don't think we are being very consistent about it. In most cases, we don't document implementation details in the doc comments.

Reply:
Agreed, it would be nice to do this, at least for new implementations.

Reply:
Ok, I'll move reference of the algorithm to the doc comment.